sdk/identity/Azure.Identity/src/ManagedIdentitySource.cs

```diff
@@ -154,9 +154,14 @@ private static AccessToken GetTokenFromResponse(in JsonElement root)
                 }
             }
 
-            return accessToken != null && expiresOn.HasValue
-                ? new AccessToken(accessToken, expiresOn.Value)
-                : throw new AuthenticationFailedException(AuthenticationResponseInvalidFormatError);
+            if (accessToken != null && expiresOn.HasValue)
+            {
+                return new AccessToken(accessToken, expiresOn.Value, InferManagedIdentityRefreshInValue(expiresOn.Value));
```

Comment:
is it possible that the MI endpoint returns expires on or expires in? If so should we defer to that instead of calculating it?

Reply:
I'm not sure, but I'd prefer to address that in a different PR if so.